libraries/CurieBLE/src/internal/BLEDeviceManager.h

```diff
@@ -392,7 +400,7 @@ class BLEDeviceManager
     uint8_t    _peer_adv_data[BLE_MAX_ADV_BUFFER_CFG][BLE_MAX_ADV_SIZE];
     uint8_t    _peer_adv_data_len[BLE_MAX_ADV_BUFFER_CFG];
     uint8_t    _peer_scan_rsp_data[BLE_MAX_ADV_BUFFER_CFG][BLE_MAX_ADV_SIZE];
-    int8_t     _peer_scan_rsp_data_len[BLE_MAX_ADV_BUFFER_CFG];
+    uint8_t    _peer_scan_rsp_data_len[BLE_MAX_ADV_BUFFER_CFG];
```

Comment:
This change has been considered. You implementation has an assumption, the scan response should followed by advertisement. But this may has some corner case to make ADV not be reported to sketch.
Now I use timeout mechanism. The central will report ADV data after 800ms delay.